src/System.Collections/tests/Generic/HashSet/HashSet.Generic.Tests.cs

```diff
@@ -340,5 +341,22 @@ public void CanBeCastedToISet()
             ISet<T> iset = (set as ISet<T>);
             Assert.NotNull(iset);
         }
+
+        [Fact]
+        public void GetObjectData_ComparerEqualsType_Success()
+        {
+            var hashSet = new HashSet<string>();
+            SerializationInfo testData = new SerializationInfo(typeof(HashSet<string>), new FormatterConverter());
+            hashSet.GetObjectData(testData, new StreamingContext(StreamingContextStates.Other));
+
+            foreach (SerializationEntry entry in testData)
+            {
+                if (entry.Name == "Comparer")
```

Comment:
Do we not expect this type check to be true of all of our types and all of their serialization entries? i.e. why not make this test just part of the larger run-this-check-for-every-type-in-the-suite set of tests?

Reply:
> Do we not expect this type check to be true of all of our types and all of their serialization entries?

No. That's why I made it local and scoped it to only this one entry. 

As mentioned there are several legit ways to pass in a value in that's not assignable by the supplied type. If it's worth it I can experimentally add that check to the test suite to find out how many types are doing this.

Reply:
> If it's worth it I can experimentally add that check to the test suite to find out how many types are doing this.

I think it's worth it.  Even if we believe the docs were wrong, they were reasonable, and thus it's quite likely there are other such dependencies.  Let's understand how widespread the issue actually is.